exchanges/huobi/huobi_websocket.go

```diff
@@ -125,13 +128,16 @@ func (h *HUOBI) wsFunnelConnectionData(ws stream.Connection, url string) {
 
 // wsReadData receives and passes on websocket messages for processing
 func (h *HUOBI) wsReadData() {
-	h.Websocket.Wg.Add(1)
 	defer h.Websocket.Wg.Done()
 	for {
-		resp := <-comms
-		err := h.wsHandleData(resp.Raw)
-		if err != nil {
-			h.Websocket.DataHandler <- err
+		select {
+		case <-h.Websocket.ShutdownC:
+			return
```

Comment:
I found that cli was not reenabling because it was stuck with a comms channel that still had an entry and no way to process it. Please use the following here to avoid failures when re-enabling a websocket:
```
case <-h.Websocket.ShutdownC:
			select {
			case <-comms:
			default:
			}
			return
```
You may also wish to use `err := h.wsHandleData(resp.Raw)` when draining the channel to not have any data loss